eeg_flow/tasks/epochs_evoked.py

```diff
@@ -256,9 +262,32 @@ def _create_epochs_evoked_and_behavioral_metadata(
             metadata=metadata_reponse,
             reject=None,
             preload=True,
-            baseline=(None, 0),
+            baseline=None,  # manual baseline
             picks="eeg",
         )
+        # at this point, the only epochs dropped are due to bad segments, let's now
+        # apply the mean baseline correction using the stimulus-locked baseline period.
+        distance = np.full(
+            (events[:, 0].size, epochs_response.events[:, 0].size),
+            epochs_response.events[:, 0],
+            dtype=float,
+        )
+        distance = distance.T - events[:, 0]  # (n_stimuli, n_response)
+        distance[np.where(distance <= 0)] = np.nan
+        idx = np.nanargmin(distance, axis=1)
+        tf = events[idx, 0]  # stimuli onset of the preceding stimulus
+        delta = int(np.round(raw.info["sfreq"] * 0.2))  # 200 ms before stimulus onset
+        t0 = tf - delta
+        baseline = np.zeros(
+            (epochs_response._data.shape[0], epochs_response._data.shape[1], delta)
+        )
+        eegs = _picks_to_idx(raw.info, "eeg", exclude=())
+        assert eegs.size == epochs_response._data.shape[1]  # sanity-check
+        for k, (t1, t2) in enumerate(zip(t0, tf, strict=True)):
```

Comment:
Hi there, something seems to be wrong here
I ran this on my ~84 (4 files for each of the 21p) files and only half work, 

To be precise it was able to output
19 files for oddball 1, 
19 for odball 2
2 for UT 1
5 for UT 2

Here's the error message, and I can send you a file if useful at this stage?

`
[3675525708:create_epochs_evoked_and_behavioral_metadata:190] ERROR: The file for participant P02, group G2, task UT, run 1 could not be processed.
[3675525708:create_epochs_evoked_and_behavioral_metadata:198] ERROR: could not broadcast input array from shape (64,0) into shape (64,205)
Traceback (most recent call last):
  File "C:\Users\anguyen\AppData\Local\Temp\ipykernel_48052\3675525708.py", line 112, in create_epochs_evoked_and_behavioral_metadata
    ) = _create_epochs_evoked_and_behavioral_metadata(raw)
  File "C:\Users\anguyen\AppData\Local\Temp\ipykernel_48052\3675525708.py", line 290, in _create_epochs_evoked_and_behavioral_metadata
    baseline[k, :] = raw[eegs, t1:t2][0]
ValueError: could not broadcast input array from shape (64,0) into shape (64,205)`

Any ideas?
Thanks!

Reply:
Can you share one of the broken files? I need to drop into a debugger console to figure out why `raw[eegs, t1:t2][0]` yields an empty array.

Reply:
here!
https://we.tl/t-5BLFo8HRT6

Reply:
I forgot that we cropped the `raw` thus `raw.first_samp` is not equal to 0.